rpc/gnmi/mixed-schema.md

```diff
@@ -108,50 +109,187 @@ if all transactions succeed should the `SetResponse` indicate success. If any
 of the transactions fail, the contents of all origins MUST be rolled back, and
 an error status returned upon responding to the `Set` RPC.
 
-## Example
+## Special considerations for a setRequest with openconfig and cli origins
+
+### Use cases
+The operational use cases for a setRequest `replace` operation containing both OC and CLI origins include:
+
+1. The ability to support incremental transition from CLI based configuration to OpenConfig configuration.
+2. A device may implement configuration items which are only defined in CLI while other configuration items are supported with OpenConfig.
+3. The desire to use a full configuration replacement strategy.
+
+To support these use cases a special meaning to a setRequest with replace origin `cli` and replace origin `openconfig` is defined.
+
+### SetRequest behavior
+
+A `SetRequest` message using the `replace` operation for both `cli` and `openconfig` origins SHOULD be performed as follows:
 
-### Example: OpenConfig and CLI Data
+1. `origin` `cli` SHOULD appear first.  
+2. The entire contents of the `ascii` encoded `cli` data SHOULD appear in a single update.
+3. The `replace` operation for origin `cli` SHOULD be interpeted as a full configuration replacement.  In other words, the target device SHOULD erase the `cli` configuration and replace it with the contents of the `cli` update message.
+4. `origin` `openconfig` should appear second.  
+5. The target device should merge the `cli` and `openconfig` and apply the merged configuration as a single replace operation, replacing the combined `cli` and `openconfig` configuration of the device.
+6. If an error occurs in merging or applying the merged configuration the device should behave as specified in [Transactionality of Sets with multiple Origins](#transactionality-of-sets-with-multiple-origins).
+
+## Examples
+
+### Example: OpenConfig and CLI origin `replace` operation
 
 If a client wishes to replace both OpenConfig and CLI-modelled data
 concurrently, it can send the following `SetRequest`:
 
-```
+```json
 replace: <
   path: <
-    origin: "openconfig"
+    origin: "cli"
   >
   val: <
-    json_val: `
-      {
-        "interfaces": {
-          "interface": [
-            {
-              "name": "eth0",
-              "config": {
-                "name": "eth0",
-                "admin-status": "UP"
-              }
-            }
-          ]
-        }
-      }
-    `
+    ascii_val: "
+    qos traffic-class 0 name target-group-BE0
+    qos tx-queue 0 name BE0"
   >
 >
 replace: <
   path: <
-    origin: "cli"
+    origin: "openconfig"
   >
   val: <
-    ascii_val: "router bgp 15169"
+    json_val: `
```

Comment:
Should be `json_ietf_val`  - there is no protocol for `json_val` here

Reply:
@earies, why `json_val` is not allowed anymore?